replica/compaction_group.hh

```diff
@@ -218,11 +218,39 @@ public:
 using storage_group_vector = utils::chunked_vector<std::unique_ptr<storage_group>>;
 
 class storage_group_manager {
+protected:
+    table& _t;
+    // The compaction group list is only a helper for accessing the groups managed by the storage groups.
+    // The list entries are unlinked automatically when the storage group, they belong to, is removed.
+    mutable compaction_group_list _compaction_groups;
```

Comment:
This mutable is wrong, IIRC I asked @raphaelsc to remove it.

Reply:
I am removing this compaction_group_list. It was only temporarily added for easy integration.